fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/TransactionsIT.java

```diff
@@ -194,6 +196,47 @@ public void testCreateDoStuffAndRollbackTransaction() throws IOException {
                 CONFLICT.getStatusCode(), getStatus(addTxTo(new HttpGet(newLocation), txLocation)));
     }
 
+    @Test
+    public void testRollbackShouldNotLeaveDbInPartiallyUpdatedState() throws IOException {
+        /* create a tx */
+        final String txLocation = createTransaction();
+
+        /* create a new object inside the tx */
+        final String newLocation;
+        final HttpPost postNew = new HttpPost(serverAddress);
+        postNew.addHeader(ATOMIC_ID_HEADER, txLocation);
+        try (final CloseableHttpResponse resp = execute(postNew)) {
+            assertEquals(CREATED.getStatusCode(), getStatus(resp));
+            newLocation = getLocation(resp);
+        }
+
+        final var resourceId = StringUtils.substringAfterLast(newLocation, "/");
+
+        /* fetch the created tx from the endpoint */
+        try (final CloseableDataset dataset = getDataset(addTxTo(new HttpGet(newLocation), txLocation))) {
+            assertTrue(dataset.asDatasetGraph().contains(ANY, createURI(newLocation), ANY, ANY));
+        }
+        /* fetch the created tx from the endpoint */
+        assertEquals("Expected to not find our object within the scope of the transaction",
+                NOT_FOUND.getStatusCode(), getStatus(new HttpGet(newLocation)));
+
+        // Add a conflicting record to the database
+        final var containmentIndex = getBean("containmentIndex", ContainmentIndex.class);
+        containmentIndex.addContainedBy("bogus", FedoraId.getRepositoryRootId(), FedoraId.create(resourceId));
+        containmentIndex.commitTransaction("bogus");
+
+        // Commit transaction -- should fail
+        assertEquals(CONFLICT.getStatusCode(), getStatus(new HttpPut(txLocation)));
+
+        assertEquals("Rolled back transaction should be gone",
+                GONE.getStatusCode(), getStatus(new HttpGet(txLocation)));
+
+        assertEquals("Expected to not find our object after rollback",
```

Comment:
I think this `NOT_FOUND` response is coming from the containment index, which failed and should have rolled back. Could we have one of the other services fail instead, like the ocfl Index, reference service or membership service? Just to ensure all the transactions are being rolled back.

Reply:
The Fedora to OCFL mapping is persisted first, and the containment index. This test does correctly confirm that the OCFL mapping is rolled back when the containment index update fails -- except in MySQL which is apparently not rolling back for some reason.

Reply:
On GET resource requests, the the OCFL mapping table is accessed when the resource headers are loaded before the containment table is checked.

Reply:
My mistake, you are correct.